### PR TITLE
Build WASM using docker

### LINKED ISSUE #239: Wasm build docker image
depends on #154

For the wasm that we want to include in the source, we should provide a docker image to build them with to get a consistent hash.

Ideally, we should be able to run a single command to build e.g. `vps/vp_user` and be able to reuse the docker image for all other wasms (with build cache, if possible).

### PR DESCRIPTION
**Pull request by Fraccaman**
Close #239

### CODE REVIEW

**Comment by Fraccaman** [comment]
Merge after #254 gets merged.

**Comment by Fraccaman** [comment]
@yito88 
- yeah, I was thinking that same. I think this can be improved by starting from a smaller initial image (at the moment the building image is ubuntu 20.04).  
- I can try using starting from `rust:latest` and check if things improve

**Comment by Fraccaman** [comment]
Using `rust:latest` it 1.55GB

**Review by yito88 — COMMENTED** [review]
Thank you for the updates!
I left some comments.

File: scripts/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+FROM rust:latest
```
[the remark this answers is not in the document]
Comment (Fraccaman) [reply scripts/Dockerfile:1]: make sense

File: Makefile [hunk 2]
```diff
@@ -73,8 +73,12 @@ doc:
 	# build and opens the docs in browser
 	$(cargo) doc --open
 
+build-wasm-scripts-docker:
+	docker run --rm  -v ${PWD}:/usr/local/rust/project anoma-wasm-2 make build-wasm-scripts
```
[the remark this answers is not in the document]
Comment (Fraccaman) [reply Makefile:77]: Ops, sorry I was testing and I left the 2 there

**Review by yito88 — APPROVED** [review]
LGTM!

**Comment by tzemanovic** [comment]
I'm just trying this and unfortunately the sccache doesn't work with wasm, because the wasm crate use `crate-type = ["cdylib"]`. It's a known limitation in sccache: https://github.com/mozilla/sccache/blob/3f318a8675e4c3de4f5e8ab2d086189f2ae5f5cf/README.md#rust :(

Also, I think having `$(rustup) toolchain install $(nightly)` in `build-wasm-scripts-docker` is not needed, the rust version is set via `rust-toolchain` file, so the nightly is not going to be used. However the one we have in the root directory has more components than what's needed to build wasm.

I'll open a follow-up to:
- remove sccache
- move all wasms into a wasm sub-directory and add a `rust-toolchain` with minimal components

**Pull request merged**